File: python/example/sort/service/pure/quicksort_service.py
import otterop.lang.array
import otterop.lang.string
import example.sort.service.quicksort_service

class QuicksortService:
    def __init__(self):
        self.otterop = example.sort.service.quicksort_service.QuicksortService()

    def sort(self, array):
        _array_array = [None] * len(array)
        for i in range(len(array)):
            _array_array[i] = otterop.lang.string.String.wrap(array[i])
        _array = otterop.lang.array.Array.wrap(_array_array)
        ret_otterop = self.otterop.sort(_array)
        ret = [None] * ret_otterop.size()
        for i in range(len(ret)):
            ret_i = ret_otterop.get(i)
            ret[i] = ret_i.unwrap()
        return ret

    def unwrap(self):
        return self.otterop

    @staticmethod
    def wrap(wrapped):
        ret = example.sort.service.pure.quicksort_service.QuicksortService.__new__(example.sort.service.pure.quicksort_service.QuicksortService)
        ret.otterop = wrapped
        return ret

File: go/example/sort/service/pure/quicksortservice.go
package pure

import (
    otteropservice "github.com/otterop/example-sort/go/example/sort/service"
    otteroplang "github.com/otterop/otterop/go/lang"
)

type QuicksortService struct {
    otterop *otteropservice.QuicksortService
}


func QuicksortServiceWrap(otterop *otteropservice.QuicksortService) *QuicksortService {
    ret := new(QuicksortService)
    ret.otterop = otterop
    return ret
}


func QuicksortServiceNew() *QuicksortService {
    ret := new(QuicksortService)
    ret.otterop = otteropservice.QuicksortServiceNew()
    return ret
}



func (this *QuicksortService) Sort(array []*string) []*string {
    var _arrayArray = make([]*otteroplang.String, len(array))
    for i := 0; i < len(array); i++ {
        _arrayArray[i] = otteroplang.StringWrap(array[i])
    }
    var _array = otteroplang.ArrayWrap(_arrayArray);
    retOtterop := this.otterop.Sort(_array)
    ret := make([]*string, retOtterop.Size())
    for i := 0; i < retOtterop.Size(); i++ {
        retI := retOtterop.Get(i)
        ret[i] = retI.Unwrap()
    }
    return ret
}

func (this *QuicksortService) Unwrap() *otteropservice.QuicksortService {
    return this.otterop
}
